cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/preview/DefaultPreviewManager.java

```diff
@@ -215,6 +219,11 @@ public ApplicationId start(@Name("namespaceId") NamespaceId namespace, AppReques
       throw new IllegalStateException("Preview service is not running. Cannot start preview for " + programId);
     }
 
+    LinkedTreeMap<String, Double> config = (LinkedTreeMap<String, Double>) previewRequest.getAppRequest().getConfig();
+    Double numOfRecords = config.get("numOfRecordsPreview");
```

Comment:
nit: avoid hardcoded strings

Reply:
Not sure how to avoid this, do I need to introduce a constant in cdap-site.xml? Seemed like an overkill and unnecessary